examples/conversation_sql/react_sql.ipynb

```diff
@@ -0,0 +1,480 @@
+    "create_sqlite_db_from_schema(db_schema_path, db_sqlite_path)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can show a general description of the database schema using the code below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The database consists of four tables: \n",
+      "\n",
+      "1. Stadium: This table includes the following fields: Stadium_ID (which is the primary key), Location, Name, Capacity, Highest, Lowest, and Average.\n",
+      "\n",
+      "2. Singer: This table includes the following fields: Singer_ID (which is the primary key), Name, Country, Song_Name, Song_release_year, Age, and Is_male (a Boolean field that indicates the singer's gender).\n",
+      "\n",
+      "3. Concert: This table is linked to the Stadium table via the Stadium_ID field. The table includes the fields concert_ID (which is the primary key), concert_Name, Theme, Stadium_ID, and Year.\n",
+      "\n",
+      "4. Singer_in_Concert: This table acts as a junction table to establish a many-to-many relationship between the Singer and Concert tables. It includes the fields concert_ID and Singer_ID, which both make up the primary key for the table. These are also reference as foreign keys to their respective basetable: the Concert table via the concert_ID field, and the Singer table via the Singer_ID field.\n",
+      "\n",
+      "In summary, this database seems designed to track concerts, including details about singers, the concert's location (stadium), and which singer performed at which concert.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from loguru import logger\n",
+    "from sql_utils import SQLPrompt\n",
+    "describe_prompt = SQLPrompt().describe_schema(db_sqlite_path)\n",
+    "sql_description = loaded_model([{\"role\": \"assistant\", \"content\": describe_prompt}]).text\n",
+    "logger.chat(sql_description)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Setting up tool functions\n",
+    "Then, we define the tools needed for React Agent to perform SQL query. Namely, our agent should be able to generate sql query given natural language and perform sql query to get the result.\n",
+    "\n",
+    "- We referenced a third-party Text-to-SQL [DAIL-SQL](https://github.com/BeachWang/DAIL-SQL) to generate Text-to-SQL prompt.\n",
+    "- We use the query_sqlite service function in the agentscope.service module. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from agentscope.service import ServiceResponse, ServiceExecStatus, ServiceFactory\n",
+    "from sql_utils import DailSQLPromptGenerator\n",
+    "\n",
+    "def generate_sql_query(question: str, db_path: str, model: Callable) -> ServiceResponse:\n",
```

Comment:
Maybe we can add this service function into `agentscope.service.text_to_sql`?

Reply:
Maybe we can integrate DAIL-SQL functions in to `agentscope.service` later.